src/reporter.ts

```diff
@@ -75,8 +75,29 @@ export const sendSpans = async (spans: any[], addEnrichmentSpan: boolean = true)
   safeExecute(logSpans)(rtt, spans);
 };
 
+const isNotJsonString = (str: string) => {
```

Comment:
A few notes on that:
1. I think that having it named `isNot` and negating the result is confusing... IMO it's better to name it `isJsonObjectOrArray`, remove the negation so it would be:
```
  return (
    (trimmed.startsWith('{') && trimmed.endsWith('}')) ||
    (trimmed.startsWith('[') && trimmed.endsWith(']'))
  );
``` 
and negate the result on the caller side.
Also, I've suggested to rename it to `isJsonObjectOrArray` since `1` or `true`, for instance, are also valid JSON values but won't be identified as such by the `{,},[,]` check. Naming it that way will express that it's focused in objects and arrays only

Reply:
1. Agree on that. With this function, I am trying to reduce the number of JSON.parse() operation